homeassistant/components/knx/weather.py

```diff
@@ -74,22 +71,19 @@ def _create_weather(xknx: XKNX, config: ConfigType) -> XknxWeather:
 class KNXWeather(KnxEntity, WeatherEntity):
     """Representation of a KNX weather device."""
 
+    _device: XknxWeather
+    _attr_temperature_unit = TEMP_CELSIUS
+
     def __init__(self, xknx: XKNX, config: ConfigType) -> None:
         """Initialize of a KNX sensor."""
-        self._device: XknxWeather
         super().__init__(_create_weather(xknx, config))
-        self._unique_id = f"{self._device._temperature.group_address_state}"
+        self._attr_unique_id = f"{self._device._temperature.group_address_state}"
```

Comment:
There is no need to f-string this?

```suggestion
        self._attr_unique_id = self._device._temperature.group_address_state
```

Reply:
inside `_device` these are `GroupAddress` objects https://github.com/XKNX/xknx/blob/a9ebe999fff118a55b96e4dba70a04101aabcd2e/xknx/telegram/address.py#L197

Reply:
In that case, I think casting them is a clearer intent in that case?

```suggestion
        self._attr_unique_id = str(self._device._temperature.group_address_state)
```